### ISSUE TITLE
Better way to query undirected graph when filtering on Edge property

### ISSUE THREAD

**Issue by phil0stine** [orientechnologies/orientdb#5540]
Here is an example of 3 people, where one pair of people currently know each other while the other pair used to.

The query I want to make is, "who are the people that Bob currently knows". Simple enough question, but the resulting query is quite messy, how could I clean it up?

Thanks

```SQL
CREATE CLASS Person EXTENDS V
CREATE CLASS knows EXTENDS E
CREATE PROPERTY Person.name STRING
CREATE PROPERTY knows.currently BOOLEAN
CREATE VERTEX Person SET name = "Bob"
CREATE VERTEX Person SET name = "Alice"
CREATE VERTEX Person SET name = "Eve"
CREATE EDGE knows FROM (SELECT FROM Person WHERE name = "Bob") TO (SELECT FROM Person WHERE name = "Alice") SET currently = true
CREATE EDGE knows FROM (SELECT FROM Person WHERE name = "Eve") TO (SELECT FROM Person WHERE name = "Bob") SET currently = false
SELECT expand(out) FROM (SELECT if(eval("out.name = 'Bob'"), inV(), outV()) AS out FROM (SELECT EXPAND(bothE('knows')) FROM Person WHERE name = "Bob") WHERE currently = true)
```

**Comment by luigidellaquila** [comment]
Hi @phil0stine 

The query should just be
```
select from (
select expand(bothE("knows")[currently = true].bothV()) from Person where name = 'Bob'
) where name <> 'Bob'
```

In v 2.2 you will also be able to express it with a MATCH statement:

```
MATCH {class:Person, where: (name='Bob'), as:bob}
        .bothE('knows'){where:(currently = true)}
        .bothV(){as:friend, where:($matched.bob != $currentMatch)}
RETURN friend
```

See docs here: https://github.com/orientechnologies/orientdb-docs/blob/master/SQL-Match.md

Thanks

Luigi

**Comment by phil0stine** [comment]
Thanks @luigidellaquila , one last question, how would the query change if we were testing for a null case instead of whether currently = true?

**Comment by luigidellaquila** [comment]
Hi @phil0stine 

unfortunately `is null` and `is not null` are not supported between square brackets (we are working to fix the problem in 2.2), so you have to use a subquery:

```
select from (
    select expand(bothV()) from (
         select expand(bothE("knows")) from Person where name = 'Bob'
    ) where currently is not null
) where name <> 'Bob'
```
